### PR TITLE
backup: add design doc of stream backup

### PR DESCRIPTION
**Pull request by YuJuncen**
### What problem does this PR solve?

Issue Number: ref #29501

Problem Summary: Added the design doc about log backup.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

### Release note

```release-note
None
```

### CODE REVIEW

File: br/docs/cn/2021-12-09-TiDB-log-based-incremental-backup-draft-1.md [hunk 1]
```diff
@@ -0,0 +1,459 @@
+## Structure
```
Comment (3pointer) [inline br/docs/cn/2021-12-09-TiDB-log-based-incremental-backup-draft-1.md:1]: This document should not in `cn` directory?

**Review by 3pointer — APPROVED** [review]

File: br/docs/en/2021-12-09-TiDB-log-based-incremental-backup-draft-1.md [hunk 2]
```diff
@@ -0,0 +1,459 @@
...
+
+## Design
+
+### BR
+
+A new subcommand is added to BR to control incremental backup tasks. (Provide a TiDB HTTP API as well, see the spec.)
+
+#### Start task
+
+```shell
```
Comment (kennytm) [inline br/docs/en/2021-12-09-TiDB-log-based-incremental-backup-draft-1.md:43]:
``````suggestion
```console
``````

File: br/docs/en/2021-12-09-TiDB-log-based-incremental-backup-draft-1.md [hunk 3]
```diff
@@ -0,0 +1,459 @@
+## Structure
```
Comment (kennytm) [inline br/docs/en/2021-12-09-TiDB-log-based-incremental-backup-draft-1.md:1]: shouldn't this be placed in `/docs/design/`?

Comment (YuJuncen) [reply br/docs/en/2021-12-09-TiDB-log-based-incremental-backup-draft-1.md:1]: The structure of this document seems doesn't match the template (lacking of test design, alternative, etc..).

(However, the template doesn't be applied for many docs... Maybe just replace `Structure` with `Introduction` , add a TOC and replace `Design` with `Detailed Design` would be fine?)

File: docs/design/2021-12-09-TiDB-log-based-incremental-backup-draft-1.md [hunk 4]
```diff
@@ -0,0 +1,495 @@
+# TiDB Log-based Incremental Backup
+
+- Author(s): @kennytm
```
Comment (kennytm) [inline docs/design/2021-12-09-TiDB-log-based-incremental-backup-draft-1.md:3]: 👀

File: docs/design/2021-12-09-TiDB-log-based-incremental-backup-draft-1.md [hunk 5]
```diff
@@ -0,0 +1,495 @@
...
+log unless the TxnEntry scan proved to have a really bad effect on
+performance.
+
+-   Scan the raft log directly. The current raftstore implementation needs to be adjusted, such as raft log gc. In addition, it is also necessary to ensure that the scan is performed during the leader’s lifetime, otherwise the complete data may not be seen.
+
+-   Also need to pay attention to region epoch check, raft log commit != apply successfully
+
+-   To support the interpretation of raftlog, I think I need to implement most of the logic of raftstore now, especially some operations such as split/merge/conf change. Due to the existence of these operations, the original total order raft log in a region has a sloping relationship between different regions. I think it is not easy to solve. This is also the reason why cdc did not use raftlog or wal at the time.
+
+## Trivia
```
Comment (kennytm) [inline docs/design/2021-12-09-TiDB-log-based-incremental-backup-draft-1.md:473]: Parts that only relevant during discussion don't have to be included...

File: docs/design/2021-12-09-TiDB-log-based-incremental-backup-draft-1.md (hunk 4 above)
Comment (YuJuncen) [reply docs/design/2021-12-09-TiDB-log-based-incremental-backup-draft-1.md:3]: yep, most of the content is just copy & paste from *TiDB Log-based Incremental Backup (Draft 1)*...

File: docs/design/2021-12-09-TiDB-log-based-incremental-backup-draft-1.md (hunk 5 above)
Comment (YuJuncen) [reply docs/design/2021-12-09-TiDB-log-based-incremental-backup-draft-1.md:473]: Which means we should remove the `trivia` part?

Comment (kennytm) [reply docs/design/2021-12-09-TiDB-log-based-incremental-backup-draft-1.md:473]: yes

Comment (YuJuncen) [reply docs/design/2021-12-09-TiDB-log-based-incremental-backup-draft-1.md]: done

**Review by joccau — APPROVED** [review]
LGTM

**Pull request merged**